### PR TITLE
fix: compressor didn't chunkify big payload

### PR DESCRIPTION
**Pull request by g11tech**
As per snappy frame encoding format, an uncompressed chunk can't be greater than `UNCOMPRESSED_CHUNK_SIZE = 65536`.
This PR updates the same in the compressor and includes a test case for the same, as well as bumps the version.

### CODE REVIEW

File: lib/compress-stream.js [hunk 1]
```diff
@@ -52,18 +60,25 @@ CompressStream.prototype._uncompressed = function (chunk) {
 
 CompressStream.prototype._transform = function (chunk, enc, callback) {
   var self = this
+  async function compressChunks() {
```
Comment (dapplion) [inline lib/compress-stream.js:63]: Why async function?

File: lib/compress-stream.js [hunk 2]
```diff
@@ -52,18 +60,25 @@ CompressStream.prototype._uncompressed = function (chunk) {
...
+    try {
+      for (let startFrom = 0; startFrom < chunk.length; startFrom += UNCOMPRESSED_CHUNK_SIZE) {
+        const endAt = startFrom + Math.min(chunk.length - startFrom, UNCOMPRESSED_CHUNK_SIZE);
+        const bytesChunk = chunk.slice(startFrom, endAt);
+        const compressed = snappy.compressSync(bytesChunk)
```
Comment (dapplion) [inline lib/compress-stream.js:68]: Why switch to the sync version? If we want to do that we should do it in another PR as study in depth the implications

File: lib/compress-stream.js (hunk 1 above)
Comment (g11tech) [reply lib/compress-stream.js:63]: the API of transform is to call callback on the completion of transformation of the chunk

**Comment by mpetrunic** [comment]
I suggest merging https://github.com/ChainSafe/node-snappy-stream/pull/4 first ^^

File: lib/compress-stream.js (hunk 2 above)
Comment (g11tech) [reply lib/compress-stream.js:68]: @dapplion  the async version where I chunk bytes from outside (like call compressStream.write(chunk)) which uses the original flow, compared to this where chunking happen in the transform using this way is 2X better:
image
(orig-snappy is the original)

If I use async version inside with `await Promise((resolve)=>...` wrap to make sure the async returned data is written in a proper serial order, its 10-5% slower because of the overhead, if I don't care about the serial order in which chunks are written (which leads to incorrect stream), then async chunking version inside transform is 5% better.

I think for now this is our best bet :slightly_smiling_face:

File: lib/compress-stream.js (hunk 1 above)
Comment (dapplion) [reply lib/compress-stream.js:63]: But there is no await in this body, and `compressChunks()` is called without any handling. Errors will become unhandled rejections. Can you just make it
```ts
  function compressChunks() {
```

File: lib/compress-stream.js (hunk 2 above)
Comment (dapplion) [reply lib/compress-stream.js:68]: Okay sounds good to me. Can you commit a benchmark tho? To inform us of the cost of compressing an uncompressing objects of 10Kb, 100Kb, 1000Kb

File: lib/compress-stream.js [hunk 3]
```diff
@@ -50,20 +58,40 @@ CompressStream.prototype._uncompressed = function (chunk) {
...
+ *  | Size              | in transform | compressStream.write | orginal (no chunking) |
+ *  |-------------------|--------------|----------------------|-----------------------|
+ *  | 10kb   (1 chunk)  | 0.0229 ms    | 0.0385 ms            | 0.0388 ms             |
+ *  | 100kb  (2 chunks) | 0.0562 ms    | 0.1051 ms            | 0.0844 ms             |
+ *  | 1000kb (16 chunks)| 0.382  ms    | 0.7971 ms            | 0.1998 ms             |
+ * 
+ */
+
+CompressStream.prototype._transform = function(chunk, enc, callback) {
   var self = this
```
Comment (wemeetagain) [inline lib/compress-stream.js:77]: I guess we don't need self anymore now

**Review by wemeetagain — APPROVED** [review]
LGTM
